### PR TITLE
MAINT: Fix chain exception in npyio.py

### LINKED ISSUE #15986: Chain exceptions where appropriate
Picking up from #15731, there are many places in numpy where we do something like:
```python
try:
    something_which_throws_typeError
except TypeError:
    raise ValueError("a clearer explanation of what went wrong")
```

It would produce a marginally clearer error if we could change these to use traceback chaining. Our two options are either:

1. The inner error provides valuable information that is not present in the outer error. We should use `from e`.
    ```python
    try:
        something_which_throws_typeError
    except TypeError as e:
        raise ValueError("a clearer explanation of what went wrong") from e
    ```
2. The inner error provides  no valuable information that is not present in the outer error. We should use `from None`:
   ```python
   try:
       some_dict[name]
   except KeyError:
       raise ValueError("The name %s is not supported" % name) from None
   ```
   In that example all the information from the `KeyError` is already in the message of the new error, so there is no point chaining

An example of such a change would be [this line](https://github.com/numpy/numpy/pull/15731/files#diff-e55dc29434893b7f529192a2e77e9b07R7070) of this otherwise-discarded patch.

For now, I would recommend we only apply this to cases where the exception is thrown unconditionally, as other cases can be more nuanced.

---

### Please see [this list](https://github.com/numpy/numpy/issues/15986#issuecomment-660762674) for potential places to contribute. Special thanks to @nkleinbaer for compiling the list!

### PR DESCRIPTION
**Pull request by bijeshmohan**
This solution fixes #15986 
@eric-wieser can you review this PR?

Thank you

### CODE REVIEW

**Comment by eric-wieser** [comment]
This has merge conflicts with master - can you `git merge` to pull in the changes?

**Comment by bijeshmohan** [comment]
@eric-wieser, I am a n00b. Both to git and github and this is my first contribution. Can you be more specific about what I need to do now? Sorry for the trouble.

**Comment by eric-wieser** [comment]
If I click on your branch name, I see "This branch is 1 commit ahead, 1152 commits behind numpy:master." One of the 1152 changes made since you created your branch edits the same lines as this PR.

You need to run `git merge` in your command line, deal with the conflicts, and then run `git push`

**Comment by bijeshmohan** [comment]
I am just messed up. Closing this PR!

**Comment by eric-wieser** [comment]
Let me see if I can fix this from the web UI

File: numpy/lib/npyio.py [hunk 1]
```diff
@@ -906,6 +906,21 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
...
+            fh = np.lib._datasource.open(fname, 'rt', encoding=encoding)
+            fencoding = getattr(fh, 'encoding', 'latin1')
+            fh = iter(fh)
+            fown = True
+        else:
+            fh = iter(fname)
+            fencoding = getattr(fname, 'encoding', 'latin1')
+    except TypeError as e:
+        raise ValueError('fname must be a string, file handle, or generator') from e
+
```
Comment (eric-wieser) [inline numpy/lib/npyio.py:923]: I assume this was a merge conflict
```suggestion
```

**Pull request closed without merging**